### DISCUSSION TITLE
Можно ли понять причину вылета при совмещении модов?

### DISCUSSION THREAD

**Discussion by Darkkys** [OGSR/OGSR-Engine#405] [Q&A]
[_________CRASH OGSM ARTIFACT.txt](https://github.com/OGSR/OGSR-Engine/files/7914815/_________CRASH.OGSM.ARTIFACT.txt)
Попробовал запустить огсм на огср-движке.. Совместил дублирующиеся файлы, вроде в основном все работает (на старте), но есть стабильный краш вот с таким логом.. (до краша в логе ничего особенного нет(ошибок\предупреждений), кроме ранее упомянутого спама в лог от последней версии движка)

**Comment by xrSimpAI** [comment]
Мне сложно представить, что нужно сделать, чтобы получить здесь ошибку

https://github.com/OGSR/OGSR-Engine/blob/f940dc5f5b1e4fe72f268fec636e16133f47ef51/ogsr_engine/xrGame/Artifact.cpp#L415

**Reply by Darkkys** [reply]
[_________CRASH OGSM ARTIFACT.txt](https://github.com/OGSR/OGSR-Engine/files/7918178/_________CRASH.OGSM.ARTIFACT.txt)
Краш при взятии в руки артефакта. Сам он берется (появляется на экране в руках), но стоит закрыть инвентарь - краш.
На всякий скинул полный лог.

**Reply by xrSimpAI** [reply]
попробуй артефакту поменять name, name_short

**Reply by Darkkys** [reply]
Это где и на что менять? )

**Reply by xrSimpAI** [reply]
```ini
inv_name				= gunsl_wpn_knife
inv_name_short			= gunsl_wpn_knife
```

**Reply by Darkkys** [reply]
Хм, помогло. Получается такое всем артефактам нужно сделать?

Другой вопрос (спрошу тут же) - при выходе из игры выскакивает такой лог
[____LOG.txt](https://github.com/OGSR/OGSR-Engine/files/7918423/____LOG.txt)

**Reply by AndreyKharkov** [reply]
> Хм, помогло. Получается такое всем артефактам нужно сделать?

Дополню от себя. Связка OGSR (2.1.359) + OGSM (Версия 2.4.3 Gold Edition)
Дело было в отсутствии значения в параметре inv_name_short. Я просто сдублировал туда значение inv_name Вылет возникал при попытке использования артефакта в пистолетном слоте.

**Comment by xrSimpAI** [comment]
> Хм, помогло.

А что там было то?

А по последнему логу, судя по куче таких сообщений
```
[22.01.22 20:34:35.727] !! [CGameObject::lua_game_object] you are trying to use a destroyed object [77f2db20]
[22.01.22 20:34:35.727] !! [CScriptGameObject::object] You are trying to use a destroyed object [77f2db20]
[22.01.22 20:34:35.728] !! [CGameObject::lua_game_object] you are trying to use a destroyed object [77f2db20]
```
Какой-то ужас в xr_kamp скорее всего. Возьми норм скрипт откуда нибудь. В твоем видимо всё совсем плохо.

**Reply by Darkkys** [reply]
Он не мой - он от огсм'а )
Если его убрать (оставив ванильный) - игра крашит на загрузке

В той строке (конкретно - проверял на медузе) в нэйме было af-medusa, в шортнейме пусто

**Reply by AndreyKharkov** [reply]
> Он не мой - он от огсм'а ) Если его убрать (оставив ванильный) - игра крашит на загрузке

Привет! Я аналогичным занимаюсь - пытаюсь прикрутить рендер OGSR к OGSM (Версия 2.4.3 Gold Edition).
Ты его победил? У меня тоже с таким же глюком вылетает при попытке "взять" артефакт. Пока он стоит в слоте - всё ном, стоит по "1" "вооружится" им - вылет с таким же логом.

**Reply by AndreyKharkov** [reply]
> Какой-то ужас в xr_kamp скорее всего. Возьми норм скрипт откуда нибудь. В твоем видимо всё совсем плохо.

Я дополню предыдущего "страдальца". Этот лог при выходе появляется не всегда. Могу примерно набросать схему: Начал новую игру (всё ок) - У Волка взял первое задание (бандиты на АТП), но не стал его выполнять - ушёл за мост, собрал там всё, дошёл до выхода с локации (вынес братков) - вернулся в домик (где потом будет Лис) - тут меня настигло выбросом, переждал его - (с логом всё ок) - сохранился контрольно и пошёл назад, к АТП - как только перешел насыпь и немного приблизился к АТП, тут и началась эта срань в логе при выходе из игры. 

"xr_kamp" - И откуда его взять? У меня такие же ошибки с OGSR (2.1.359) + OGSM (Версия 2.4.3 Gold Edition)

**Reply by xrSimpAI** [reply]
Из какого-нибудь аи пака например. Последние версии ai additions вроде нормальные должны быть, естественно не 2009-х годов, которые в большинстве модов.

**Reply by AndreyKharkov** [reply]
Я тебя ещё не достал? :) Помоги, плиз. Я немного разобрался с этим вылетом и сранью в логе. Дело в том, что в OGSM (Версия 2.4.3 Gold Edition) добавлен мод "Dynamic Campfire Mod" (динамические костры). Методом "научного тыка" и чтением логов я выяснил - если рядом на локации нет костров - всё в порядке и лог чистый. Если рядом есть костры, то в логе начинается эта срань: [CGameObject::lua_game_object] you are trying to use a destroyed object [4b1f0300] и далее по логу
[xray_reanimator_04-02-22_03-16-11.log](https://github.com/OGSR/OGSR-Engine/files/7999341/xray_reanimator_04-02-22_03-16-11.log)
Движку не нравится вот это: xr_kamp.script:1074: in function 'decreasePops' , это как раз из функции гашения костров function CKampManager:decreasePops(npc) из "Dynamic Campfire Mod"
Я понимаю, что это наглость, но можешь помочь, глянуть, что там не сопрягается с движком? Вот файл: xr_kamp (кстати, он 2018 года, родной от OGSM)
[xr_kamp.zip](https://github.com/OGSR/OGSR-Engine/files/7999359/xr_kamp.zip)

И ещё, попутно, где взять такой инвентарь ГГ, как на первом скрине вот тут: ?
https://github.com/OGSR/OGSR-Engine/releases/tag/2.1.181

**Reply by xrSimpAI** [reply]
чел, я же тебе выше написал, возьми нормальный хр_камп. В нормальном хркампе нету динамических костров. Им там и не место. Я тебе и без смотрения скрипта скажу, там скорее всего клиентские объекты в таблице хранят, чего делать нельзя. Где-то я такое уже видел.

**Reply by xrSimpAI** [reply]
> И ещё, попутно, где взять такой инвентарь ГГ, как на первом скрине вот тут:

в НС

**Reply by AndreyKharkov** [reply]
Да, ты прав, там клиентские объекты хранят в этом же скрипте в таблице...
... только в той-же НС динамические костры как раз и есть и как раз в этом скрипте, там там у функций "CKampManager:increasePops" и "CKampManager:decreasePops(npc)" такие полотна кода.....
Эх, вот бы кто помог.... Хоть подскажи, что с этим делать?.... Только по аналогии с солянкой?...

**Reply by xrSimpAI** [reply]
В НС скрипт ужасен. Я выше написал что делать.

**Reply by AndreyKharkov** [reply]
Ок. Благодарю. Я наконец-то скачал оригинал этого мода костров, проще его вырезать и всё! Ибо я в душе не ипу, будет ли он работать, если его оформить отдельным скриптом. Я не на столько асс в этом, как ты понял...

P.S. После удаления из OGSM (Версия 2.4.3 Gold Edition) "Dynamic Campfire Mod" срань из логов и вылеты пропали.

**Comment by AndreyKharkov** [comment]
Новые нестыковки OGSR (2.1.359) + OGSM (Версия 2.4.3 Gold Edition)
Вылет происходит, видимо, при попытке NPC бросить в меня гранату. Ниже - лог.
[xray_reanimator_01-02-22_00-54-24.log](https://github.com/OGSR/OGSR-Engine/files/7974783/xray_reanimator_01-02-22_00-54-24.log)
[xrs_grenade.zip](https://github.com/OGSR/OGSR-Engine/files/7974803/xrs_grenade.zip)


Не подскажете, что это означает? Что ему не хватает?

**Reply by I-AmDead** [reply]
Ля, чувак, у тебя просто нету секции **grenade_f1_test**, ты предлагаешь так каждый вылет не относящийся к движку за тебя разбирать?

**Reply by AndreyKharkov** [reply]
Эта секция есть вот в этом файле:
[weapons.zip](https://github.com/OGSR/OGSR-Engine/files/7975103/weapons.zip)
Чувак, этот вылет появился после замены движка в OGSM (Версия 2.4.3 Gold Edition) я хочу прикрутить туда этот движок. и не я один!

**Reply by I-AmDead** [reply]
Учись значит читать логи чтобы понимать что игра у тебя просит, тогда у тебя отпадут такие банальные вопросы.

**Reply by I-AmDead** [reply]
У тебя видимо в конфигах не подключен grenadier.ltx или у него всратая кодировка

**Reply by AndreyKharkov** [reply]
Чувак, ты конечно чувак, но спасибо! Дело было в структуре weapons.ltx , #include "grenadier.ltx" должно было быть в конце этого файла, ниже секции [explosive_barrel]. Видимо оригинальный движок пропускал как-то эту ошибку...
